### ISSUE TITLE
Birday does not properly handle Feb 29 birthdays

### ISSUE THREAD

**Issue by IdmFoundInHim** [m-i-n-a-r/birday#35]
Birday crashes upon launch or any interaction when the time is set to 00:07 on 1 Jan 2021. It can be fixed by setting the date to 31 Dec 2020. Experienced on a OnePlus 7 Pro running LineageOS  for microG 17.1 (Dec update; based on Android 10 Open Source Project)

**Comment by m-i-n-a-r** [comment]
Well that's kinda weird, how can I recreate this? I have your same exact phone and it works as intended, at least now (11.50 in my time zone)

**Comment by IdmFoundInHim** [comment]
I found the problem in logcat; it appears that a birthday entry for Feb 29 is causing the problem when it tries to project that birthday onto the current year.

```log
01-01 08:14:21.523 12875 12911 E AndroidRuntime: Process: com.minar.birday, PID: 12875
01-01 08:14:21.523 12875 12911 E AndroidRuntime: java.lang.RuntimeException: Exception while computing database live data.
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at j.t.l$a.run(:4)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:919)
01-01 08:14:21.523 12875 12911 E AndroidRuntime: Caused by: java.time.format.DateTimeParseException: Text '2021-02-29' could not be parsed: Invalid date 'February 29' as '2021' is not a leap year
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.format.DateTimeFormatter.createError(DateTimeFormatter.java:1920)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1855)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.LocalDate.parse(LocalDate.java:394)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.LocalDate.parse(LocalDate.java:379)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at b.b.a.g.g.a(Unknown Source:4)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at b.b.a.g.f.call(:7)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        ... 4 more
01-01 08:14:21.523 12875 12911 E AndroidRuntime: Caused by: java.time.DateTimeException: Invalid date 'February 29' as '2021' is not a leap year
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.LocalDate.create(LocalDate.java:423)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.LocalDate.of(LocalDate.java:263)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.chrono.IsoChronology.resolveYMD(IsoChronology.java:560)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.chrono.IsoChronology.resolveYMD(IsoChronology.java:123)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.chrono.AbstractChronology.resolveDate(AbstractChronology.java:472)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.chrono.IsoChronology.resolveDate(IsoChronology.java:492)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.chrono.IsoChronology.resolveDate(IsoChronology.java:123)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.format.Parsed.resolveDateFields(Parsed.java:351)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.format.Parsed.resolveFields(Parsed.java:257)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.format.Parsed.resolve(Parsed.java:244)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.format.DateTimeParseContext.toResolved(DateTimeParseContext.java:331)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1955)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
01-01 08:14:21.523 12875 12911 E AndroidRuntime:        ... 8 more
```

**Comment by m-i-n-a-r** [comment]
You're perfectly right, I completely forgot to manage the 29 February case! My fault D:
I'll fix it asap!

**Comment by m-i-n-a-r** [comment]
Fixed in 6ece7a6a1f120aee9d9c977baa4b9100c662fad8 , i'll release the update in the next 24 hours! Thanks again!
